skbio/io/fastq.py

```diff
@@ -480,21 +480,30 @@ def _parse_sequence_data(fh):
 def _parse_quality_scores(fh, seq_len, variant, phred_offset):
     phred_scores = []
     qual_len = 0
-    for chunk in _line_generator(fh):
-        if chunk.startswith('@') and qual_len == seq_len:
-            return phred_scores, chunk
+    prev_was_blank = False
+    for chunk in fh:
+        chunk = chunk.strip()
```

Comment:
Is it possible to factor this new code to use `_line_generator`? That way there's only a single place where line preprocessing (i.e., whitespace stripping) takes place. Adding a flag to `_line_generator` allowing/disallowing blank lines should do the trick.